test/Dialect/FIRRTL/parse-basic.fir

```diff
@@ -1111,6 +1111,19 @@ circuit MyModule :     ; CHECK: firrtl.circuit "MyModule" {
     ; CHECK-NEXT: [[NOT_COND:%.+]] = firrtl.not %cond
     ; CHECK-NEXT: firrtl.assert %clock, [[NOT_COND]], [[NOT_ENABLE]]
 
+    ; Verification Library Covers
+
+    ; Predicate modifier `noMod`
+    when not(cond):
+      printf(clock, enable, "Assertion failed: [verif-library-cover]<extraction-summary>{\"predicateModifier\":{\"type\":\"noMod\"},\"conditionalCompileToggles\":[{\"type\":\"unrOnly\"},{\"type\":\"formalOnly\"}],\"labelExts\":[\"cover\",\"label\"],\"format\":{\"type\":\"sva\"},\"baseMsg\":\"cover hello world\"}", value)
+      stop(clock, enable, 1)
+    ; CHECK: [[TMP_INV:%.+]] = firrtl.not %cond
+    ; CHECK: [[TMP:%.+]] = firrtl.not [[TMP_INV]]
+    ; CHECK-NOT: firrtl.assert %clock, [[TMP]], %enable, "cover hello world"(%value) : !firrtl.uint<42>
```

Comment:
@seldridge is this the right place to check that the assert is removed?

Reply:
Some of the other checks put it somewhere else. However, this example would want to use a `firrtl.assert` and not a `firrtl.stop`. (This check will obviously pass if there is no input `firrtl.assert` in the test!)

The easiest way to check that this does the right thing would be to copy this test to a standalone file, then delete the lines around [here](https://github.com/llvm/circt/blob/main/lib/Dialect/FIRRTL/Import/FIRParserAsserts.cpp#L325), and make sure that the test then fails.

Reply:
cool, moving to `.assert` and commenting out the `opsToErase` did fail. I will keep the `CHECK-NOT` where it is then, unless you think there is a better place for it.

Reply:
ok wait the `CHECK-NOT` is passing all the time... If i just do `firtool -parse-only foo.fir` with my printf + assert, the assert definitely stays in there :(

Reply:
never mind, the assert is getting removed, I was confusing myself with my bad print statements